bokehjs/src/coffee/models/canvas/cartesian_frame.coffee

```diff
@@ -45,8 +45,10 @@ export class CartesianFrame extends LayoutCanvas
   _get_scales: (scale, ranges, frame_range) ->
     scales = {}
     for name, range of ranges
-      if range.type == 'Range1d' or range.type == "DataRange1d"
-        s = if scale? then scale else new LinearScale()
+      if scale?
+        s = scale
+      else if range.type == 'Range1d' or range.type == "DataRange1d"
+        s = new LinearScale()
       else if range.type == "FactorRange"
         s = new CategoricalScale()
       else
```

Comment:
As you pointed out this weirdness in the earlier PR. I'd say this is another reason why BokehJS should just always expect a real scale. There's no way for it to know to use a log scale without either sending the scale itself, or a hinting of some sort (what we were trying to get rid of).